### ISSUE TITLE
Using as a library

### ISSUE THREAD

**Issue by alexpineda** [meszaros-lajos-gyorgy/node-pkware#1]
Do you have an example of this being used within another node application?

I'm attempting to do so myself but running into issues where `implode` actually creates buffers larger than the input. Explode is working great except for one edge case where I am expecting 8192 bytes back but get half of that, 4096. Thanks for your efforts!

Example below is from my nearly completed project: https://github.com/darkmatter-gg/downgrade-replay

```javascript
const deflate = async (buf) =>
  new Promise((res, rej) => {
    const t = new Transform({
      transform: implode(BINARY_COMPRESSION, DICTIONARY_SIZE3),
    });

    new Readable({
      read: function () {
        this.push(buf);
        this.push(null);
      },
    })
      .pipe(t)
      .pipe(
        new Writable({
          write(def, enc, done) {
            res(def);
            done();
          },
        })
      );
  });
```

**Comment by meszaros-lajos-gyorgy** [comment]
Hi! Thank you very much for trying out the code. Your code works absolutely correctly and what you've experienced is actually correct, because I didn't yet finished the implode algorithm implementation. The implode algorithm has 2 main features: first it will turn the data into a bitstream and that part have been implemented fully, while the second part looks for repetitions in a certain sized buffer and this is the part, which I haven't finished yet. The second part is what actually does the compression.

I'll try to finish it as soon as I can, but in the mean time the imploded output will actually be bigger compared to the original file.

There is a ticket for the compression on my public trello board, where you can follow the progress: https://trello.com/c/r6GlKIXt/1-implement-compression-for-implode

**Comment by alexpineda** [comment]
wow. ok thank you!

**Comment by alexpineda** [comment]
Would I be able to get your email or discord tag if you have one?

**Comment by meszaros-lajos-gyorgy** [comment]
Hello! You can find me as `Lali (NodenGaming)#7802` on Discord.

**Comment by alexpineda** [comment]
Thanks. I tried (NodenGaming)#7802 and NodenGaming#7802 to no avail. It is case sensitive so maybe there is some other capitalization?

**Comment by meszaros-lajos-gyorgy** [comment]
Hi! Sorry for the very long delay, but now the development of node-pkware is back on it's track and I've released the first beta version for 1.0, where there is proper library support for explode: https://www.npmjs.com/package/node-pkware/v/1.0.0-beta.0

Implode is not in this version, but explode is fully functional and if you are still interested, then you could check, if the code meets your requirements.

**Comment by meszaros-lajos-gyorgy** [comment]
https://github.com/meszaros-lajos-gyorgy/node-pkware/releases/tag/v1.0.0 have been relesed, which can finally compress data resulting in a smaller output then the input.
